auto_dev/utils/rollback.py

```diff
@@ -0,0 +1,65 @@
+"""Filesystem utilities for temporary backups and rollback mechanisms."""
+
+import shutil
+import signal
+import tempfile
+from pathlib import Path
+from contextlib import chdir, contextmanager
```

Comment:
is it possible to not use this chidr? 

iirc os.chdir can be used as a context as well, and then it means we dont drop an entire python version